pkg/controller/controller_test.go

```diff
@@ -1811,9 +1811,60 @@ func TestProvisionFromSnapshot(t *testing.T) {
 }
 
 // TestProvisionWithTopology is a basic test of provisioner integration with topology functions.
-func TestProvisionWithTopology(t *testing.T) {
+func TestProvisionWithTopologyEnabled(t *testing.T) {
 	defer utilfeaturetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.Topology, true)()
 
+	const requestBytes = 100
+
+	testcases := map[string]struct {
+		driverSupportsTopology bool
+		nodeLabels             []map[string]string
+		topologyKeys           []map[string][]string
+		expectedNodeAffinity   *v1.VolumeNodeAffinity
+		expectError            bool
+	}{
+		"topology success": {
+			driverSupportsTopology: true,
+			nodeLabels: []map[string]string{
+				{"com.example.csi/zone": "zone1", "com.example.csi/rack": "rack1"},
+				{"com.example.csi/zone": "zone1", "com.example.csi/rack": "rack2"},
+			},
+			topologyKeys: []map[string][]string{
+				{driverName: []string{"com.example.csi/zone", "com.example.csi/rack"}},
+				{driverName: []string{"com.example.csi/zone", "com.example.csi/rack"}},
+			},
+			expectedNodeAffinity: &v1.VolumeNodeAffinity{
+				Required: &v1.NodeSelector{
+					NodeSelectorTerms: []v1.NodeSelectorTerm{
+						{
+							MatchExpressions: []v1.NodeSelectorRequirement{
+								{
+									Key:      "com.example.csi/zone",
+									Operator: v1.NodeSelectorOpIn,
+									Values:   []string{"zone1"},
+								},
+								{
+									Key:      "com.example.csi/rack",
+									Operator: v1.NodeSelectorOpIn,
+									Values:   []string{"rack2"},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		"topology fail": {
+			driverSupportsTopology: true,
+			topologyKeys: []map[string][]string{
+				{driverName: []string{"com.example.csi/zone", "com.example.csi/rack"}},
+				{driverName: []string{"com.example.csi/zone", "com.example.csi/rack"}},
+			},
+			expectError: true,
+		},
+		"driver doesn't support topology": {},
```

Comment:
nit: kind of odd to just have an empty test case. maybe just explicitly set some of the important fields to false. `driverSupportsTopology` and `expectError` I guess.